lib/rabbit_feed/event_definitions.rb

```diff
@@ -51,19 +51,31 @@ def defined_as &block
         @definition = block.call if block.present?
       end
 
-      def payload
-        ([
+      def payload_schema
+        { name: "#{name}_payload", type: 'record', fields: fields.map(&:schema) }
+      end
+
+      def metadata_schema
+        { name: 'event_metadata', type: 'record', fields: [
           (Field.new 'application',    'string', 'The name of the application that created the event'),
           (Field.new 'host',           'string', 'The hostname of the server on which the event was created'),
           (Field.new 'environment',    'string', 'The environment in which the event was created'),
-          (Field.new 'version',        'string', 'The version of the event'),
+          (Field.new 'version',        'string', 'The version of the event payload'),
+          (Field.new 'gem_version',    'string', 'The version of rabbit_feed by which the event was created'),
```

Comment:
:smile: I don't understand what I wrote myself.

I meant... (and maybe still won't make any sense). The fact that the message structure changes is not linked to the fact that the rabbit feed gem version changes. So the fact that now the event has metadata should not be linked to the gem... I mean when the structure of the event changes, like now, the Event version changes, when the gem_version changes it doesn't affect the event version..

So we would have `payload_version`, `event_version`, `gem_version`... but don't know if the gem_version is needed inside the event